loki-build-image/Dockerfile

```diff
@@ -1,28 +1,35 @@
-FROM golang:1.11.4-stretch
-RUN apt-get update && apt-get install -y file jq unzip protobuf-compiler libprotobuf-dev libsystemd-dev && \
-	rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
-ENV DOCKER_VER="17.03.0-ce"
-RUN curl -L -o /tmp/docker-$DOCKER_VER.tgz https://download.docker.com/linux/static/stable/x86_64/docker-$DOCKER_VER.tgz && \
-	tar -xz -C /tmp -f /tmp/docker-$DOCKER_VER.tgz && \
-	mv /tmp/docker/* /usr/bin && \
-	rm /tmp/docker-$DOCKER_VER.tgz
-ENV HELM_VER="v2.13.1"
-RUN curl -L -o /tmp/helm-$HELM_VER.tgz http://storage.googleapis.com/kubernetes-helm/helm-${HELM_VER}-linux-amd64.tar.gz && \
-	tar -xz -C /tmp -f /tmp/helm-$HELM_VER.tgz && \
-	mv /tmp/linux-amd64/helm /usr/bin/helm && \
-	rm -rf /tmp/linux-amd64 /tmp/helm-$HELM_VER.tgz
+FROM alpine as helm
+ARG HELM_VER="v2.13.1"
+RUN apk add --no-cache curl && \
+    curl -L -o /tmp/helm-$HELM_VER.tgz http://storage.googleapis.com/kubernetes-helm/helm-${HELM_VER}-linux-amd64.tar.gz && \
+    tar -xz -C /tmp -f /tmp/helm-$HELM_VER.tgz && \
+    mv /tmp/linux-amd64/helm /usr/bin/helm && \
+    rm -rf /tmp/linux-amd64 /tmp/helm-$HELM_VER.tgz
+
+FROM alpine as golangci
+RUN apk add --no-cache curl && \
+    cd / && \
+    curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s v1.17.1
+
+FROM golang:1.11.4-alpine
```

Comment:
I looked around to see what Alpine packages have systemd headers and it isn't great: 

https://pkgs.alpinelinux.org/contents?file=sd-journal.h&path=&name=&branch=edge

Probably possible to download them and put them where they need to go.

Is the penalty for using debian the download time of the build image? and would we just incur that once per build?

Reply:
Hrm, see the point, but it's `1.01GB` vs `573MB`.:
```
loki                                 latest                       acd7110a51d1        2 minutes ago       1.01GB
loki-alpine                          latest                       4e9a4fd95781        4 minutes ago       573MB
```

Reply:
And no, @slim-bean, this would happen for every step, because how CircleCI works: every time `setup_remote_docker` happens, we receive a new daemon which has to re-pull.

Reply:
Okay, the current one is compress 500mb, with some optimizations I can get down to 300mb. This is still bad compared to alpine, but I can live with that